packages/lit-dev-content/site/tutorials/content/working-with-lists/02.md

```diff
@@ -0,0 +1,40 @@
+The previous step showed how a `for ... of` loop can be used to populate an
+array of templates to render. However, Lit also provides a built-in
+`map` directive for a more declarative way of writing templates. It provides
+an interface similar to the `map` array prototype method for any generic
+iterables, and can make code easier to read.
+
+To solve the previous step by using it, first import the `map` directive.
+
+{% switchable-sample %}
+
+```ts
+// my-element.ts
+import {map} from 'lit/directives/map.js';
+```
+
+```js
+// my-element.js
+import {map} from 'lit/directives/map.js';
+```
+
+{% endswitchable-sample %}
+
+Then use the directive in the rendered template.
+
+{% switchable-sample %}
+
+```ts
+// my-element.ts
+${map(this.items, (item) => html`<li>${item}</li>`)}
+```
+
+```js
+// my-element.js
+${map(this.items, (item) => html`<li>${item}</li>`)}
+```
+
+{% endswitchable-sample %}
+
+The index of the item is also made available to the callback function. Check
```

Comment:
Let's try to add some guidance (re-emphasizing the point in the previous slide) about when you should do one way v. the other. E.g. something like... Use the `map` directive for simple cases that easily lend themselves to declarative description; use imperative code for more complex situations.

Reply:
Yes, that's good direction. Added.